extensions/oidc-common/runtime/src/main/java/io/quarkus/oidc/common/runtime/OidcCommonUtils.java

```diff
@@ -284,8 +284,23 @@ public static boolean isClientSecretBasicAuthRequired(Credentials creds) {
     }
 
     public static boolean isClientJwtAuthRequired(Credentials creds) {
-        return creds.jwt.secret.isPresent() || creds.jwt.secretProvider.key.isPresent() || creds.jwt.key.isPresent()
-                || creds.jwt.keyFile.isPresent() || creds.jwt.keyStoreFile.isPresent();
+        int keyMaterialConfigCount = 0;
+        if (creds.jwt.secret.isPresent() || creds.jwt.secretProvider.key.isPresent()) {
+            keyMaterialConfigCount++;
+        }
+        if (creds.jwt.key.isPresent()) {
+            keyMaterialConfigCount++;
+        }
+        if (creds.jwt.keyFile.isPresent()) {
+            keyMaterialConfigCount++;
+        }
+        if (creds.jwt.keyStoreFile.isPresent()) {
+            keyMaterialConfigCount++;
+        }
+        if (keyMaterialConfigCount > 1) {
+            throw new ConfigurationException("Only a single OIDC JWT credential key property can be configured");
```

Comment:
This error is cryptic to me. Maybe specify the extra key properties already configured that are not mutually exclusive?

Reply:
Hi @gastaldi Hopefully it looks better now, thanks :-)

Reply:
@gastaldi Actually, sorry, I'll need to tweak it further, to correctly report `quarkus.oidc-client` vs `quarkus.oidc`, sorry

Reply:
No problem, I'll keep it approved as I trust you ;)